api/src/pcapi/core/offers/repository.py

```diff
@@ -480,16 +480,17 @@ def get_offers_map_by_id_at_provider(id_at_provider_list: list[str], venue: offe
     return offers_map
 
 
-def get_offers_map_by_venue_reference(id_at_provider_list: list[str], venue_id: int) -> dict[str, int]:
+def get_offers_map_by_venue_reference(id_at_provider_list: list[str], venue_id: int) -> dict[str | None, int]:
```

Comment:
<!-- Questions are appropriate if you have a potential concern but are not quite sure if it's relevant or not. Asking the author for clarification or investigation can lead to a quick resolution.  -->
**❔ question:** ‏On peut vraiment avoir `None` comme clé ?

Reply:
Oui `compute_venue_reference` **peut** renvoyer None

Reply:
Proposition: ajouter un petit `# TODO` pour jeter un oeil à ce comportement.
Je pense que la valeur clé `None` apporte pas de valeur à la feature ..
J'approuve pour le typage :slightly_smiling_face:

Reply:
Je suis pas fan de mettre un TODO dans le code en sachant pertinemment qu’il ne sera jamais traité. J’ai été un peu plus loin du coup, `compute_venue_reference` n’est appelé qu’à deux endroit, ici et dans `synchronize_provider_api`. Ici le premier argument qu’on lui passe ne peut pas être None parce que `id_at_provider_list` est typé comme `list[str]`, et dans `synchronize_provider_api` le typage de retour `str | None` posait problème (ce qui fait un type ignore supplémentaire d’enlevé) parce qu’on passe le résultat en paramètre où l’on attend seulement une string. Bref, en conclusion, en l’état actuel de la code base, on prend toujours une string et on retourne une string.